torch_em/model/unetr.py

```diff
@@ -0,0 +1,296 @@
+        return masks
+
+    def forward(self, x):
+        org_shape = x.shape[-2:]
+
+        x = torch.stack([self.preprocess(e) for e in x], dim=0)
+
+        z0 = self.z_inputs(x)
+
+        z12, from_encoder = self.encoder(x)
+        x = self.base(z12)
+
+        from_encoder = from_encoder[::-1]
+        z9 = self.deconv1(from_encoder[0])
+
+        z6 = self.deconv1(from_encoder[1])
+        z6 = self.deconv2(z6)
+
+        z3 = self.deconv1(from_encoder[1])
+        z3 = self.deconv2(z3)
+        z3 = self.deconv3(z3)
+
+        updated_from_encoder = [z9, z6, z3]
+        x = self.decoder(x, encoder_inputs=updated_from_encoder)
+        x = self.deconv4(x)
+        x = torch.cat([x, z0], dim=1)
+
+        x = self.decoder_head(x)
+
+        x = self.out_conv(x)
+        x = self.final_activation(x)
+
+        x = self.postprocess_masks(x, org_shape, org_shape)
+        return x
+
+
+class SingleDeconv2DBlock(nn.Module):
+    def __init__(self, in_planes, out_planes):
+        super().__init__()
+        self.block = nn.ConvTranspose2d(in_planes, out_planes, kernel_size=2, stride=2, padding=0, output_padding=0)
+
+    def forward(self, x):
+        return self.block(x)
+
+
+class SingleConv2DBlock(nn.Module):
+    def __init__(self, in_planes, out_planes, kernel_size):
+        super().__init__()
+        self.block = nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=1,
+                               padding=((kernel_size - 1) // 2))
+
+    def forward(self, x):
+        return self.block(x)
+
+
+class Conv2DBlock(nn.Module):
+    def __init__(self, in_planes, out_planes, kernel_size=3):
+        super().__init__()
+        self.block = nn.Sequential(
+            SingleConv2DBlock(in_planes, out_planes, kernel_size),
+            nn.BatchNorm2d(out_planes),
+            nn.ReLU(True)
+        )
+
+    def forward(self, x):
+        return self.block(x)
+
+
+class Deconv2DBlock(nn.Module):
+    def __init__(self, in_planes, out_planes, kernel_size=3):
+        super().__init__()
+        self.block = nn.Sequential(
+            SingleDeconv2DBlock(in_planes, out_planes),
+            SingleConv2DBlock(out_planes, out_planes, kernel_size),
+            nn.BatchNorm2d(out_planes),
+            nn.ReLU(True)
+        )
+
+    def forward(self, x):
+        return self.block(x)
+
+
+def build_unetr_with_sam_intialization(out_channels=1, model_type="vit_b", checkpoint_path=None):
+    if get_sam_model is None:
+        raise RuntimeError(
+            "micro_sam is required to initialize the UNETR image encoder from segment anything weights."
+            "Please install it from https://github.com/computational-cell-analytics/micro-sam"
+            "and then rerun your code."
+        )
+    predictor = get_sam_model(model_type=model_type, checkpoint_path=checkpoint_path)
+    _image_encoder = predictor.model.image_encoder
+
+    image_encoder = ViTb_Sam()
+    # FIXME this doesn't work yet because the parameters don't match
```

Comment:
@anwai98 I updated the code a bit compared to what you had (I am pretty sure your version wasn't yet working but didn't try it). 
But my current version still isn't working...

Reply:
Ah, okay! The last I checked, the parameters were getting initialized from SAM in the old setup (checked it before and after the function was taking charge to initialize the ViT, the encoder blocks were updating) (however, I think your implementation is smarter and we could keep it to our micro-sam standard)

Reply:
This is still not fully working for me, but I will go ahead and merge and we can fix it in a follow up.